toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/diskutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/file"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+func enableVerityPartition(imageChroot *safechroot.Chroot) error {
+	var err error
+
+	// Integrate systemd veritysetup dracut module into initramfs img.
+	systemdVerityDracutModule := "systemd-veritysetup"
+	err = buildDracutModule(systemdVerityDracutModule, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	// Update mariner config file with the new generated initramfs file.
+	err = updateMarinerCfgWithInitramfs(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func buildDracutModule(dracutModuleName string, imageChroot *safechroot.Chroot) error {
+	var err error
+
+	listKernels := func() ([]string, error) {
+		var kernels []string
+		// Use RootDir to get the path on the host OS
+		bootDir := filepath.Join(imageChroot.RootDir(), "boot")
+		files, err := filepath.Glob(filepath.Join(bootDir, "vmlinuz-*"))
+		if err != nil {
+			return nil, err
+		}
+		for _, file := range files {
+			kernels = append(kernels, filepath.Base(file))
+		}
+		return kernels, nil
+	}
+
+	kernelFiles, err := listKernels()
+	if err != nil {
+		return fmt.Errorf("failed to list kernels: %w", err)
+	}
+
+	if len(kernelFiles) == 0 {
```

Comment:
Just do 

if len(kernelFiles) != 1 {
  return fmt.Errorf("expected one kernel file, but found %d", len(kernelFiles))
}

Like you do in updateMarinerCfgWithInitramfs